File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int>f;
        for (int c:magazine){
            f[c]++;
        }
        for(int c:ransomNote){
            if(f[c]==0){
                return false;
            }
            f[c]--;
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        r={}
        m={}
        for c in(ransomNote):
            r[c]=r.get(c,0)+1
        
        for c in(magazine):
            m[c]=m.get(c,0)+1
        
        for key,value in r.items():
            if m.get(key,0)<value:
                return False
        return True